src/api/spec/jobs/report_to_scm_job_spec.rb

```diff
@@ -98,5 +110,88 @@
         subject
       end
     end
+
+    context 'when the event subscription payload contains filters' do
+      let(:event_subscription) do
+        EventSubscription.create(token: token,
+                                 user: user,
+                                 package: package,
+                                 receiver_role: 'reader',
+                                 payload: { scm: 'github', workflow_filters: workflow_filters },
+                                 eventtype: 'Event::BuildSuccess',
+                                 channel: :scm)
+      end
+
+      before do
+        event
+        event_subscription
+      end
+
+      context 'which filter out the architectures of the event' do
+        let(:workflow_filters) do
+          { architectures: { only: ['i586'] } }
+        end
+
+        let(:event) { Event::BuildSuccess.create(project: project.name, package: package.name, repository: 'openSUSE_Tumbleweed', arch: 'x86_64', reason: 'foo') }
+
+        it_behaves_like 'the job performed'
+        it_behaves_like 'not reporting to the SCM'
+      end
+
+      context 'matching the repositories of the event' do
```

Comment:
This context is testing that repositories aren't matching the event's repository:
```suggestion
      context 'not matching the repositories of the event' do
```